sdk/src/main/java/io/opentelemetry/sdk/metrics/Batcher.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.metrics;
+
+import io.opentelemetry.metrics.LabelSet;
+import io.opentelemetry.sdk.metrics.aggregator.Aggregator;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import java.util.List;
+
+/**
+ * A {@code Batcher} represents an internal representation of an {code Instrument} aggregation
+ * process. It records individual measurements (via the {@code Aggregator}). It batches together
+ * {@code Aggregator}s for the similar {@code LabelSet}.
+ */
+interface Batcher {
+
+  /**
+   * Returns the {@link Aggregator} that should be used by the bindings, or observers.
+   *
+   * @return the {@link Aggregator} used to aggregate individual events.
+   */
+  Aggregator getAggregator();
+
+  /**
+   * Batches multiple entries together that are part of the same metric. It may remove labels from
+   * the LabelSet and merge aggregations together.
+   *
+   * @param labelSet the {@link LabelSet} associated with this {@code Aggregator}.
+   * @param aggregator the {@link Aggregator} used to aggregate individual events for the given
+   *     {@code LabelSet}.
+   * @param mappedAggregator {@code true} if the {@code Aggregator} is still in used by a binding.
+   *     If {@code false} the {@code Batcher} can reuse the {@code Aggregator} instance.
+   */
+  void batch(LabelSet labelSet, Aggregator aggregator, boolean mappedAggregator);
+
+  /**
+   * Stops the current collection cycle and returns the list of metrics batched in this Batcher.
+   *
+   * <p>There may be more than one MetricData in case a multi aggregator is configured.
+   *
+   * <p>Based on the configured options this method may reset the internal state to produce deltas,
+   * or keep the internal state to produce cumulative metrics.
+   *
+   * @return the list of metrics batched in this Batcher.
+   */
+  List<MetricData> stopCollection();
```

Comment:
This method name seems funny to me. Is this called once per collection cycle, or does this turn off batching? With this name, I would assume that it was end of the lifecycle, but from reading the code, it seems like this gets called once per "harvest". If it's the latter, then this should probably be named something like "completeCollectionCycle" or something like that. What do you think, @bogdandrutu ?

Reply:
done.